public/src/feed/feed-mixin.html

```diff
@@ -81,15 +81,28 @@
           // parse route and fetch
           let parsedRoute = this._parseRoute(route);
           this._parsedRoute = parsedRoute;
+          this._moreAvailable = false;
+          this._promotedCardIds = [];
           if (parsedRoute) {
             this._onDataFetch(parsedRoute);
-            return $core.getFeed(parsedRoute.feedName, parsedRoute.count, parsedRoute.cardId, parsedRoute.channel).then((cards) => {
+            return $core.getFeed(parsedRoute.feedName, parsedRoute.count, parsedRoute.cardId, null, parsedRoute.channel, this._promotedCardIds).then((feed) => {
+              const cards = feed.cards;
+              this._lastCardId = null;
+              for (const card of cards) {
+                if (card.promoted) {
+                  this._promotedCardIds.push(card.id);
+                } else {
+                  this._lastCardId = card.id;
+                }
+              }
               this._setItems(cards);
               if (cards && cards.length) {
                 this.$.noItems.classList.add("hidden");
               } else {
                 this.$.noItems.classList.remove("hidden");
               }
+              this._moreAvailable = feed.moreAvailable && this._lastCardId ? true : false;
+              this._onMoreUpdated(this._moreAvailable);
```

Comment:
For flags that update the UI like this one, I usually add an observer to _moreAvailable -> _onMoreUpdated so we don't have to call this every time _moreAvailable is changed. 
Granted I have not been very consistent with that pattern either.